abk7777/gfe-db
# Basic build and load functionality

Basic infrastructure to build and load `gfe-db` into a running Neo4j server:
- AWS Batch Job Definitions, Queues and Compute Environments for build and load services
- StepFunctions workflow to run build and load sequentially
- Uploads logs in addition to CSV dataset
- Tested with and without alignments
- Built and loaded versions 3450 and 3440, loading 3430 right now
- Refactored repo into 3 basic components each with its own Dockerfile: 
  - Database
  - Build service
  - Load service

To Do:
- Use Fargate with AWS Batch for the load service instead of EC2 to save cost
- Create nested cloudformation templates
- Add CI/CD for Docker images
- Add trigger for when a new IMGT/HLA version is released
- Before uploading CSVs, the S3 prefix should be cleared of previously built data.
- Load script can be optimized
  - Better logging
  - Clean up the Cypher script to avoid Neo4j errors
